src/routes/tools/local-climate-change-snapshot/_select-location/SelectLocation.svelte

```diff
@@ -5,18 +5,32 @@
   import LocationMap from "./LocationMap.svelte";
   import LocationForm from "./LocationForm.svelte";
 
-  let searchValue =
-    $locationStore && $locationStore.location
-      ? $locationStore.location.title
-      : "";
-  let selectedLocation = $locationStore.location;
-  let selectedRadio = $locationStore.boundaryId || "address";
-
-  $: console.log("$locationStore: ", $locationStore);
-
-  function handleSelectLocation({ detail }) {
-    locationStore.updateLocation(detail);
-    locationStore.updateBoundary(selectedRadio);
+  const { location, boundary } = locationStore;
+
+  let searchValue = $location ? $location.title : "";
+  let selectedLocation = $location;
+  let selectedRadio = $boundary ? $boundary.id : "locagrid";
+
+  $: handleSelectBoundaryId(selectedRadio);
+  $: handleSelectLocation(selectedLocation);
+
+  $: {
+    console.log("$locationStore update: ", $locationStore);
+  }
+
+  function handleSelectLocation(value) {
+    locationStore.updateLocation(value);
+  }
+
+  function handleSelectBoundaryId(value) {
+    locationStore.updateBoundary(value);
+  }
+
+  function handleMapClick({ detail }) {
```

Comment:
Should clicking on the map in an area where there is no location (e.g. the Pacific Ocean) clear the selected location and search value?

Reply:
Inclined to say no, given that this may happen accidentally and be an annoying or confusing UX.